libnmstate/nm/nmclient.py

```diff
@@ -49,6 +51,13 @@ def nm_version():
     return NM.Client.get_version(client())
 
 
+def libnm_nm_version_mismatch_check():
```

Comment:
something needs to call this function to ensure that the check will actually happen, maybe when the client is created, @ffmancera what do you think?

Reply:
I think it would be fine to check this when the client is created. Before doing anything else to avoid unexpected errors.